csrc/expr_evaluator.cpp

```diff
@@ -141,6 +141,13 @@ void ExpressionEvaluator::bind_(
             i);
         bind_(
             rfactor_domain[i]->expandedExtent(), t.size(i), evaluate_validate);
+      } else if (
+          rfactor_domain[i]->isDeviceDim() &&
+          rfactor_domain[i]->getMaybeExpandedExtent()->isConst()) {
```

Comment:
nit: I think `getMaybeExpandedExtent()` can be replaced by `extent()` here (because we are in the else branch)